packages/libllvm/build.sh

```diff
@@ -85,13 +85,6 @@ termux_step_host_build() {
 }
 
 termux_step_pre_configure() {
-	# Version guard to keep flang in sync
```

Comment:
Shouldn't this have a mutual `TERMUX_PKG_ALIGN_VERSION_WITH=flang` to coincide with Flang's `TERMUX_PKG_ALIGN_VERSION_WITH=libllvm`?
I wouldn't expect any issues from such a "loop" in alignment.

And if that does cause an issue, it should be handled by the alignment solver.

It should be visible from every package that has an alignment with another package that that alignment exists.
Just as every package in an update group should visibly be part of said update group.

Reply:
`TERMUX_PKG_ALIGN_VERSION_WITH` handling code check package referenced in this variable and all packages referencing current package in their `TERMUX_PKG_ALIGN_VERSION_WITH` if any. So it is kinda being checked both ways.

Update group does not mean all packages in update group must have same versions. It only means they should be updated in single push/pull_request to avoid dependency update hell.

Reply:
Ofc you can do that to make it obvious, but it is not required. Anyway when you will try to update one of them without updating another one you will get version mismatch error.

Reply:
Please cross-reference it for all packages that we're adding this to at the inception of the feature so we don't have to track these down later.
I'm even considering adding a linter rule to enforce full cross-referencing of alignments so this doesn't become a mess to figure out.

> [!NOTE]
> Also in case you didn't catch it earlier,
> Turning `TERMUX_PKG_ALIGN_VERSION_WITH` into a comma separated list
> like `TERMUX_PKG_LICENSE` would be helpful since more than 2 packages may need to have their versions aligned.

Reply:
Probably it will be better to make checking code to find all packages referencing same package in `TERMUX_PKG_ALIGN_VERSION_WITH` to make sure version guard works for all of them.

Otherwise we might end up with referencing 60+ packages in every 60 buildscripts of them. And it will end up with a tree wide updating buildscripts if we add more kframeworks6 libraries.

> like TERMUX_PKG_LICENSE would be helpful since more than 2 packages may need to have their versions aligned.

I solved this by mentioning same package in all of packages requiring version aligning.

Reply:
> Probably it will be better to make checking code to find all packages referencing same package in `TERMUX_PKG_ALIGN_VERSION_WITH` to make sure version guard works for all of them.
> 
> Otherwise we might end up with referencing 60+ packages in every 60 buildscripts of them. And it will end up with a tree wide updating buildscripts if we add more kframeworks6 libraries.

That's where the update group comes in though, no?
It doesn't imply version alignment, but it does imply a grouping.

> > like TERMUX_PKG_LICENSE would be helpful since more than 2 packages may need to have their versions aligned.
> 
> I solved this by mentioning same package in all of packages requiring version aligning.

Thanks. I feel like this is gonna be a mess otherwise so I wanna get ahead of it.

Reply:
> That's where the update group comes in though, no? It doesn't imply version alignment, but it does imply a grouping.

Yep. `TERMUX_PKG_ALIGN_VERSION_WITH` should come in separate PR, but it changes same files so I not decided if I should move it away from this PR. And also it helps us make sure nobody will break packages with randoms PRs updating specific packages despite of existing (or not yet existing) version guards.